include/deal.II/trilinos/nox.templates.h

```diff
@@ -0,0 +1,1027 @@
+       * convergence.
+       */
+      template <typename VectorType>
+      class NOXCheck : public NOX::StatusTest::Generic
+      {
+      public:
+        /**
+         * Constructor.
+         */
+        NOXCheck(const std::function<SolverControl::State(const unsigned int,
+                                                          const double,
+                                                          const VectorType &,
+                                                          const VectorType &)>
+                   check_iteration_status)
+          : check_iteration_status(check_iteration_status)
+          , status(NOX::StatusTest::Unevaluated)
+        {}
+
+        /**
+         * Check status.
+         */
+        NOX::StatusTest::StatusType
+        checkStatus(const NOX::Solver::Generic &problem,
+                    NOX::StatusTest::CheckType  checkType) override
+        {
+          if (checkType == NOX::StatusTest::None)
+            {
+              status = NOX::StatusTest::Unevaluated;
+            }
+          else
+            {
+              if (check_iteration_status == nullptr)
+                {
+                  status = NOX::StatusTest::Converged;
+                }
+              else
+                {
+                  // unwrap the various vectors
+                  const VectorType &x__ = *dynamic_cast<
+                    const internal::NOXWrappers::Vector<VectorType> *>(
+                    &problem.getSolutionGroup().getX());
+                  const VectorType &f__ = *dynamic_cast<
+                    const internal::NOXWrappers::Vector<VectorType> *>(
+                    &problem.getSolutionGroup().getF());
+
+                  // forward to the user-provided function and checks
+                  // convergence
+                  const auto state = this->check_iteration_status(
+                    problem.getNumIterations(), f__.l2_norm(), x__, f__);
+
+                  // translate the returned value back to Trilinos data
+                  // structure
+                  switch (state)
+                    {
+                      case SolverControl::iterate:
+                        status = NOX::StatusTest::Unconverged;
+                        break;
+                      case SolverControl::failure:
+                        status = NOX::StatusTest::Failed;
+                        break;
+                      case SolverControl::success:
+                        status = NOX::StatusTest::Converged;
+                        break;
+                      default:
+                        AssertThrow(false, ExcNotImplemented());
+                    }
+                }
+            }
+
+          return status;
+        }
+
+        /**
+         * Return last return value of checkStatus().
+         */
+        NOX::StatusTest::StatusType
+        getStatus() const override
+        {
+          return status;
+        }
+
+        /**
+         * Print last return value of print().
```

Comment:
```suggestion
         * Print the last value that was given by `checkStatus()`
```
Note: `print()` -> `checkStatus()`